src/network.cpp

```diff
@@ -206,4 +206,32 @@ namespace net {
 
     return mapped_port;
   }
+
+  /**
+   * @brief Returns a string for use as the instance name for mDNS.
+   * @return Hostname-based instance name or "Sunshine" if hostname is invalid.
+   */
+  std::string
+  mdns_instance_name() {
+    std::string hostname = boost::asio::ip::host_name();
```

Comment:
If this became an argument for the function, the unit testing could be parametrized with various hostnames to check if it will work all the scenarios provided.

Reply:
Added in the latest commit.